test/unit/test-amp-story-player.js

```diff
@@ -691,6 +691,88 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
       return expect(() => player.go(-1)).to.throw('Out of Story range.');
     });
 
+    it('last story call to next_() is the first story', async () => {
+      const playerEl = win.document.createElement('amp-story-player');
+      appendStoriesToPlayer(playerEl, 5);
+
+      const player = new AmpStoryPlayer(win, playerEl);
+
+      await player.load();
+      await nextTick();
+
+      const navigationSpy = env.sandbox.spy();
+      playerEl.addEventListener('navigation', navigationSpy);
+      playerEl.setAttribute('enable-circular-wrapping', true);
+      player.go(4);
+      player.next_();
```

Comment:
Let's avoid calls to private methods. Can you use any of the other options we have? (swiping, `go()`, etc...)

Reply:
The swipes don't seem to trigger navigation and I have a test that checks go's circular wrapping. So I think the two options I have would be to either check iframe positions or to just get rid of these specific tests.

Reply:
The fact that the swipe doesn't trigger a navigation means this code is not doing what we want yet. A swipe in the last story should take the user to the first story in the player.

This currently doesn't work in your PR because we have a check on the horizontal swipe to check if there's a secondary iframe that we need to animate (in `getSecondaryIframe_()`, `if (nextStoryIdx < 0 || nextStoryIdx >= this.stories_.length)`. 

https://github.com/ampproject/amphtml/blob/ce4569ee8d6561c59d625e1b8f84d542be84a8aa/src/amp-story-player/amp-story-player-impl.js#L988-L998

https://github.com/ampproject/amphtml/blob/ce4569ee8d6561c59d625e1b8f84d542be84a8aa/src/amp-story-player/amp-story-player-impl.js#L1038-L1046

A way we could do this is by replacing the `delta > TOGGLE_THRESHOLD_PX && this.getSecondaryIframe_()` to `delta > TOGGLE_THRESHOLD_PX && (this.getSecondaryIframe_() || this.circularWrapping_)` here:

https://github.com/ampproject/amphtml/blob/ce4569ee8d6561c59d625e1b8f84d542be84a8aa/src/amp-story-player/amp-story-player-impl.js#L989